api/controllers/PostController.js

```diff
@@ -392,6 +400,36 @@ module.exports = {
       }
     })
     .then(() => res.ok({}), res.serverError)
-  }
+  },
+
+  checkFreshnessForCommunity: function (req, res) {
+    var foundPosts
+    var resolve = value => foundPosts = value
```

Comment:
but i need to call findPosts with the specific query for the case where
there is a NEW post id. Which is likely the most common case.

On Mon, Feb 29, 2016 at 4:26 PM, Lawrence Wang <notifications@github.com>
wrote:

> In api/controllers/PostController.js
> <https://github.com/Hylozoic/hylo-node/pull/93#discussion_r54504293>:
>
> > @@ -392,6 +400,36 @@ module.exports = {
> >        }
> >      })
> >      .then(() => res.ok({}), res.serverError)
> > -  }
> > +  },
> > +
> > +  checkFreshnessForCommunity: function (req, res) {
> > +    var foundPosts
> > +    var resolve = value => foundPosts = value
>
> the way that seems most straightforward to me is that you send as
> parameters the list of post ids you have along with the updated_at
> timestamp for each. then all you need to do on the server side is Post.query().where('id',
> 'in', thoseIds).filter(p => p.get('updated_at') > originalUpdatedAts[p.id
> ]) (pseudocode, but i hope you get the idea) and return true if that list
> is non-empty.
>
> then you avoid using findPosts at all, because you don't need to build
> the original query from scratch, and you don't need any of the presentation
> logic.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Hylozoic/hylo-node/pull/93/files#r54504293>.
>



-- 
Robbie Carlton

719.358.3482
<robbie@authenticmanprogram.com>